### ISSUE TITLE
Won't show email registration message

### ISSUE THREAD

**Issue by dottjt** [danschultzer/pow_assent#58]
I'm not 100% sure if this issue is related to pow_assent or just pow, however I've noticed it since using pow_assent.

Essentially, when I register a new user via email via /registration/new and then when I'm taken to /session/new, it doesn't show a message to the user to check their email.

You can test this here: https://neverfapdeluxe.com/registration/new

/session/new is located here:
https://github.com/dottjt/nfd-app/blob/production/lib/nfd_web/templates/pow/session/new.html.eex

**Comment by danschultzer** [comment]
The [error flash should be set](https://github.com/danschultzer/pow_assent/blob/3f6f4136ad6d7502ff6fbbb74440c3f612dfcf01/test/pow_assent/phoenix/controllers/authorization_controller_test.exs#L168). Let me do a few tests here to see what's going wrong.

**Comment by dottjt** [comment]
<img width="420" alt="スクリーンショット 2019-04-07 午前5 57 51" src="https://user-images.githubusercontent.com/8922214/55674611-24900f80-58fa-11e9-8340-c310bbd6aa46.png">

The :error flash is set, but it doesn't seem to be returning anything. I believe actually I've reported a similar issue in the past, but for whatever reason it worked when I went to test it back then. Now I can definitely say that it does not work.

**Comment by danschultzer** [comment]
I'm pretty sure it's because of the routes. What happens if you set the `after_registration_path/1` route like this:

```elixir
def after_registration_path(conn), do
  if Pow.Plug.current_user(conn),
    do: Routes.dashboard_path(conn, :dashboard),
    else: Routes.pow_session_path(conn, :new)
end
```

**Comment by danschultzer** [comment]
In short, the `Pow.Phoenix.PlugErrorHandler` module overrides the error message because user isn't signed in. By default this error message is empty so it'll just set the error flash as empty.

**Comment by dottjt** [comment]
Excellent, that seems to have done the trick. Thank you!

**Comment by danschultzer** [comment]
Great! I've fixed it upstream so when pow 1.0.5 is released it'll no longer override the error flash if the message is nil and you can remove the `after_registration_path/1` method 😄

**Comment by danschultzer** [comment]
You can remove the `after_registration_path/1` method now with pow `1.0.5` 😄
